src/test/java/de/rabea/ui/RealConsoleTest.java

```diff
@@ -12,17 +12,28 @@
     @Test
     public void readsAGivenStringFromTheConsole() {
         InputStream input = new ByteArrayInputStream("hello\n".getBytes());
-        RealConsole realConsole = new RealConsole(input, null);
+        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(input));
+        RealConsole realConsole = new RealConsole(bufferedReader, null);
         assertEquals("hello", realConsole.readUserInput());
     }
 
     @Test
     public void printsAGivenStringToTheConsole() {
         InputStream input = new ByteArrayInputStream("0\n".getBytes());
+        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(input));
         OutputStream output = new ByteArrayOutputStream();
         PrintStream printedOutput = new PrintStream(output);
-        RealConsole realConsole = new RealConsole(input, printedOutput);
+        RealConsole realConsole = new RealConsole(bufferedReader, printedOutput);
         realConsole.print("hey");
         assertEquals("hey\n", output.toString());
     }
+
+    @Test(expected=UncheckedIOException.class)
+    public void throwsAnExceptionForBadInput() throws IOException {
+        InputStream input = new ByteArrayInputStream("hello\n".getBytes());
+        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(input));
+        bufferedReader.close();
+        RealConsole realConsole = new RealConsole(bufferedReader, null);
+        assertEquals("hello", realConsole.readUserInput());
```

Comment:
Ah yes, I just took out the assertion and just call the method and the test is still green. I'll update it and remove the assertion.